### PR TITLE
resolves #168 migrate CI to GitHub Actions

### LINKED ISSUE #168: Migrate away from travis-ci.org
travis-ci.**org** is going to be shut down on **December, 31**:

https://docs.travis-ci.com/user/migrate/open-source-repository-migration#frequently-asked-questions

> Q. When will the migration from travis-ci.org to travis-ci.com be completed?
> A. In an effort to ensure that all of our users - whether you build open-source, public or private repositories - receive regular feature updates, security patches and UX/UI enhancements, we are announcing that travis-ci.org will be officially closed down completely no later than December 31st, 2020, allowing us to focus all our efforts on bringing new features and fixes to travis-ci.com and all of our awesome users like yourself on the travis-ci.com domain.

> Q. What will happen to travis-ci.org after December 31st, 2020?
> A. Travis-ci.org will be switched to a read-only platform, allowing you to see your jobs build history from all repositories previously connected to travis-ci.org.

Also, note that travis-ci.**com**, while being the most logical place to move to, has **build time limits** for public repositories:

https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing

> For those of you who have been building on public repositories (on travis-ci.com, with no paid subscription), we will upgrade you to our trial (free) plan with a 10K credit allotment (which allows around 1000 minutes in a Linux environment).

Most of Asciidoctor ecosystem has already moved to GitHub Actions.

### PR DESCRIPTION
**Pull request by slonopotamus**

### CODE REVIEW

**Comment by slonopotamus** [comment]
Example run: https://github.com/slonopotamus/docker-asciidoctor/runs/1546816395?check_suite_focus=true

Obviously, you'll need to add `DOCKERHUB_SOURCE_TOKEN` and/or `DOCKERHUB_TRIGGER_TOKEN` to GitHub Actions secrets in order to make deployment work on https://github.com/asciidoctor/docker-asciidoctor/settings/secrets/actions

**Comment by slonopotamus** [comment]
Also, current repository settings require Travis build to pass successfully before PR can be merged. This cannot be satisfied anymore because I'm removing Travis build file :)

File: .github/workflows/ci.yml
Comment (dduportal) [inline .github/workflows/ci.yml:5]:
```suggestion
    runs-on: ubuntu-20.04
```
Fix the mainline release to keep a consistent behavior over time, ease contributor reproductability, and remove warnings about upcoming upgrade. WDYT?

File: .github/workflows/ci.yml
Comment (dduportal) [inline .github/workflows/ci.yml:12]:
```suggestion
          sudo apt-get install -y --no-install-recommends bats
```
Even though the GH-Actions runners are well configured, making the "non interactive" explicit is always a good help for future contributors, WDYT?

File: .github/workflows/ci.yml
Comment (dduportal) [inline .github/workflows/ci.yml:18]: What is the rationale behind this code?
WDYT about moving it into the Makefile to provide as much reusability as possible?

**Review by dduportal — CHANGES_REQUESTED** [review]
Thanks for this great work, of for your patience before having a review!

Some feedbacks have been added if it is ok for you?

File: .github/workflows/ci.yml
Comment (slonopotamus) [reply .github/workflows/ci.yml]: OK

File: .github/workflows/ci.yml
Comment (slonopotamus) [reply .github/workflows/ci.yml:2]: I believe that `push` works for both branch and tag pushes and there's no need to add more types of triggers.

File: .github/workflows/ci.yml
Comment (slonopotamus) [reply .github/workflows/ci.yml]: OK

File: .github/workflows/ci.yml
Comment (slonopotamus) [reply .github/workflows/ci.yml:18]: Current Makefile works with `CURRENT_GIT_REF` variable that it either gets externally or calculates via `git rev-parse --abbrev-ref HEAD` that returns things like `master`.

On Travis, there is `TRAVIS_BRANCH` variable that is weirdly named (because it can point to both branches and tags), but is fully equivalent to format output by `git rev-parse --abbrev-ref HEAD`.

Now, on GitHub Actions, there's no direct equivalent. GitHub Actions has `GITHUB_REF` variable, that has values like `refs/heads/master` instead of just `master`. So we need to strip `refs/heads` (or `refs/tags`) part, that's what that code does. Alternatively, I can change `CURRENT_GIT_REF` Makefile code to stop passing `--abbrev-rev`, so Makefile always works with non-abbreviated string and cuts `refs/heads`/`refs/tags` part itself.

File: .github/workflows/ci.yml
Comment (dduportal) [reply .github/workflows/ci.yml:2]: Sounds good 👍

File: .github/workflows/ci.yml
Comment (dduportal) [reply .github/workflows/ci.yml:18]: Thanks for the explanation! Yes it would be cool to move this logic to the Makefile to uncouple a bit more from the CI system. Does it look good?

**Review by dduportal — APPROVED** [review]
LGTM great job!

Approving this as my last comment is a nice to have (and not blocking the PR in any way).

**Comment by dduportal** [comment]
- The Travis CI check had been removed
- The 2 secrets have been added

=> I'm adding an empty commit to this PR to trigger a first GH action build (as it seems to require a repository's collaborator to start)

**Comment by slonopotamus** [comment]
Note that I did *not* test that upload to docker actually works, though I don't see any reasons why it shouldn't after secrets are added.

**Comment by slonopotamus** [comment]
You can't test upload in a PR because PRs do not have access to repo secrets.

**Comment by dduportal** [comment]
No problem @slonopotamus! Thank you for this excellent work. I've merged the PR  to allow testing (and first gh action setup) on master branch to benefit from the secrets.

**Comment by dduportal** [comment]
> You can't test upload in a PR because PRs do not have access to repo secrets neither on Travis nor on GitHub Actions, this is a security measure.
Our messages crossed, thanks a lot for this precision!

**Comment by slonopotamus** [comment]
Hmm... [Deploy cannot access secrets](https://github.com/asciidoctor/docker-asciidoctor/runs/1726563909#step:7:83)... I guess we need to expose secret to environment variable. A couple of minutes, will create a follow-up PR.

**Comment by slonopotamus** [comment]
I think #171 should do the trick.

**Comment by dduportal** [comment]
@slonopotamus yes, you might want to define an `env:` directive in the deploy section as underlined in https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow to expose the secret to environment

**Comment by slonopotamus** [comment]
We could also declare it a Makefile bug: Makefile doesn't return non-zero exit code when deploy fails.

**Comment by dduportal** [comment]
> We could also declare it a Makefile bug: Makefile doesn't return non-zero exit code when deploy fails.

Yes, totally! And I might be the culprit for that 😅

**Pull request merged**
